### PR TITLE
Define proto messages for watchlist creation flow

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR defines proto messages for watchlist creation flow.

### CODE REVIEW

File: buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.dependency
+
+object JavaX {
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt:29]: I am not sure a) how is this declaration related to this PR, and b) why we need it at all.

Please review.

File: model/src/main/proto/spine_examples/shareaware/notification_rule.proto [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import "spine/money/money.proto";
+
+// Represents a rule for notifying a user when a share price meets a desired price.
+message NotificationRule {
+
+  // The ID of the share that notification rule belongs to.
+  ShareId share = 1 [(required) = true];
+
+  // The price of the share that the user wants to be notified about.
+  spine.money.Money wanted_price = 2 [(required) = true];
```
Comment (armiol) [inline model/src/main/proto/spine_examples/shareaware/notification_rule.proto:47]: This looks more like a `threshold`.

File: model/src/main/proto/spine_examples/shareaware/identifiers.proto [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+syntax = "proto3";
+
+package spine_examples.shareaware;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.shareaware.spine.io";
+option java_package = "io.spine.examples.shareaware";
```
Comment (armiol) [inline model/src/main/proto/spine_examples/shareaware/identifiers.proto:34]: Basically, it would be nice to have the same package-(folder)-per-aggregate packaging as per my comment to ChatSPN.

However, while it makes sense for aggregate state, its commands and events, identifiers should stay in the "root" of this bounded context.

@Vladyslav-Kuksiuk see this comment as well, as you don't have `identifiers.proto` yet, but eventually you'll do.

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev please see my comments.

File: buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt (hunk 1 above)
Comment (Artem-Semenov-dev) [reply buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt:29]: I am not sure but I think we need it because `Spine` using `@javax.annotation.Generated("by Spine Model Compiler")` this annotation in generated code comes from the `JavaX` library.
image
I have discovered that we may need the `JavaX` library in this PR because we generate some code here.
Anyway, I am not sure about this, please correct me if I am wrong.

Comment (armiol) [reply buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt:29]: I have this type as a part of JDK 8. It's in my `rt.jar`:
Screenshot 2023-03-08 at 12 52 28

Comment (Artem-Semenov-dev) [reply buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/JavaX.kt]: Yes, the problem was in JDK, sorry for that.

File: buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/Money.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.dependency
+
+object Money {
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/Money.kt:29]: I suggest we make this dependency description a nested object under `Spine`.

**Review by armiol — APPROVED** [review]
LGTM with a comment to address prior to merging.
